src/Player.cpp

```diff
@@ -29,8 +35,8 @@ void Player::Update(const float dt)
 	if (sf::Keyboard::isKeyPressed(sf::Keyboard::S) || (sf::Keyboard::isKeyPressed(sf::Keyboard::Down)))
 		duckySprite.move(0.f, speed * dt);
 
-	collisionShape.setPosition(duckySprite.getPosition().x, duckySprite.getPosition().y);
 	KeepPlayerInBounds();
+	collisionShape.setPosition(duckySprite.getPosition().x, duckySprite.getPosition().y);
```

Comment:
I think this one needs to be before KeepPlayerInBounds, because first we need to move the duck, and then correct its position if necessary. Otherwise, you will be correcting its position from the previous frame, and then immediately overriding it with the new one from this frame

Reply:
I disagree, but let's have a closer look.
In my solution, in update, first, we set the new position  of ducky sprite based on keybord input, then we check if the position is outside bounds. If so, we correct it, so that it stays in bounds and based on that corrected position we determine the new position of the collisionShape. Then in Draw() we draw the both on the screen in the exact same spot

https://user-images.githubusercontent.com/82574004/171144892-91475d49-09a1-4c9f-9101-e0924c65a368.mp4


In the code you propose, we set the new position of the sprite, same as above, but then, immidiately determine the position of collisionShape without checking if the sprite is out of bounds. Only then do we correct the position of the sprite. In KeepInBounds(), we correct the position of the sprite alone, so the collisionShape remains outside of bounds and is never moved to the correct spot. And so, in the Draw() we draw the corrected position of duckySprite, and unchecked position of collisionShape - the two won't always be in the same spot.

https://user-images.githubusercontent.com/82574004/171144903-90ca6b51-86c8-40bd-a4d8-246aa7dabd8d.mp4

I attach videos as evidence supporting my claim